src/components/menu/menu.component.js

```diff
@@ -1,8 +1,80 @@
-import React from 'react';
+import React, { useRef, useCallback, useState } from 'react';
 import PropTypes from 'prop-types';
 import { StyledMenuWrapper, StyledMenuItemsWrapper, StyledMenuItem } from './menu.style';
+import Events from '../../utils/helpers/events';
 
 const Menu = ({ menuType = 'light', children }) => {
+  const menuItemsRefs = useRef(React.Children.map(children, child => child.ref || React.createRef()));
+  const actualFocusedItemIndex = useRef();
+  const [isOpenSubmenu, setIsOpenSubmenu] = useState(null);
+  const setFocusToElement = useCallback((event, index) => {
+    if (event) {
+      event.preventDefault();
+    }
+
+    actualFocusedItemIndex.current = index;
+    const isFocusedOnLastMenuItem = index === menuItemsRefs.current.length;
+    const isFocusedOnFirstMenuItem = index === -1;
+    if (isFocusedOnLastMenuItem) {
+      actualFocusedItemIndex.current = 0;
+    } else if (isFocusedOnFirstMenuItem) {
+      actualFocusedItemIndex.current = menuItemsRefs.current.length - 1;
+    }
+
+    menuItemsRefs.current[actualFocusedItemIndex.current].current.focus();
+  }, []);
+
+  const handleKeyDown = useCallback((event, index) => {
```

Comment:
We should call the users `onKeyDown` we also need to check to see if they did `e.preventDefault()` in their `onKeyDown` if they did we should not run our `handleKeyDown`.

Reply:
`onKeyDown` has not been resolved in this file.